src/JETLS.jl

```diff
@@ -705,6 +707,44 @@ function initiate_context!(state::ServerState, uri::URI)
     nothing
 end
 
+
+function debounce(f::Function; delay::Float64=DEFAULT_DEBOUNCE_INTERVAL)
+    timer = nothing
+    function debounced_func(args...; kwargs...)
+        if timer !== nothing
+            close(timer)
+        end
+        timer = Timer(delay) do timer
+            f(args...; kwargs...)
+            timer = nothing
+        end
+    end
+    return debounced_func
+end
+
+function throttle(f::Function; delay::Float64=DEFAULT_THROTTLE_INTERVAL)
+    last_call = 0.0
+    timer = nothing
+    function throttled_func(args...; kwargs...)
+        now = time()
+        if now - last_call ≥ delay
+            f(args...; kwargs...)
+            last_call = now
+        else
+            if timer !== nothing
+                close(timer)
```

Comment:
```suggestion
        if timer !== nothing
            close(timer)
        end
        if now - last_call ≥ delay
            f(args...; kwargs...)
            last_call = now
        else
```
We would like to close the previous timer in the `then` case too.